### DISCUSSION TITLE
Text alignment not preserved on paste

### DISCUSSION THREAD

**Discussion by abecirovic3** [ueberdosis/tiptap#5417] [General Questions]
Should the text alignment be kept on paste in tiptap? I'm having a hard time preserving the alignment when I paste content that has the text-align property set.
In the screen recording, you can see that the alignment is preserved when the pasted content is a heading, but not a paragraph. I pasted in the editor in the playground of the text alignment extension https://tiptap.dev/docs/editor/extensions/functionality/textalign.

https://github.com/user-attachments/assets/38ea6cb9-3cca-4239-9342-e3aa1efb3c79

**Comment by nperez0111** [comment]
The text-align from w3 schools that you are copying from does not actually have the text-align set in the HTML, only in CSS. This would not be preserved.
If you copy from a tiptap editor and paste into another tiptap editor because the text-align is directly embedded in the html in the style attribute.

You can inspect your clipboard contents with a tool like https://evercoder.github.io/clipboard-inspector/ which would show you that the text you copied does not actually include the style attribute. Not all formatting is going to be preserved when copying and pasting things.

**Reply by abecirovic3** [reply]
Thanks for the reply, but I'm not sure if I can agree.
If we inspect the clipboard contents with the provided tool we will get the following

`<meta charset='utf-8'><span style="color: rgb(0, 0, 0); font-family: Times; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: center; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam semper diam at erat pulvinar, at pulvinar felis blandit. Vestibulum volutpat tellus diam, consequat gravida libero rhoncus ut.</span>`

Which has the text-align property set to center. And I also tried copy pasting from the editor on the text-align extension docs page to the notion-like editor demo editor and the alignment is not preserved, as shown in the screen recording.

https://github.com/user-attachments/assets/339de56d-ec53-4891-905a-2c98523dcae0

I must say, that I have tried these things with google docs and it's not working there either so it's no biggie, but I'll have to come up with something because the tester in my team will be like "ooh the text alignment is not preserved blah blah" 😅

**Reply by abecirovic3** [reply]
One thing I noticed is that if you copy more than one node that is aligned, the alignment will be preserved on all the nodes except the first

https://github.com/user-attachments/assets/6c9f339e-fc29-4a54-aec3-bac8132bbfca
